rust/diesel/src/product_api.rs

```diff
@@ -0,0 +1,63 @@
+use bigdecimal::BigDecimal;
+use rocket_contrib::json::Json;
+use serde::Serialize;
+
+use crate::db;
+use crate::product::{NewProduct, Product};
+
+#[post("/products", data = "<product>", format = "json")]
+pub fn create_product(product: Json<NewProduct>, connection: db::Connection) -> Json<Product> {
+    Json(Product::create(product.into_inner(), &connection))
+}
+
+#[derive(Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct ProductsResponse {
+    content: Vec<ProductView>,
+}
+
+#[derive(Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct ProductView {
+    pub product_id: i32,
+    pub product_name: String,
+    pub price: BigDecimal,
+}
+
+#[get("/products")]
+pub fn read_products(connection: db::Connection) -> Json<ProductsResponse> {
+    let product_views = Product::read_all(&connection)
+        .iter()
```

Comment:
Fixed

Reply:
It doesn't look like this has been fixed, at least from what I am seeing in the code. If you use `.into_iter()` instead of `.iter()` then you won't need to clone the product name and price in the map below.